src/app/api/course/route.ts

```diff
@@ -0,0 +1,56 @@
+import { PrismaClient } from '@prisma/client';
+import { NextResponse, NextRequest } from 'next/server';
+import { courseSchema } from '@/schemas';
+
+const prisma = new PrismaClient();
+
+export async function GET() {
+  try {
+    const courses = await prisma.course.findMany();
+    return NextResponse.json({ data: courses }, { status: 200 });
+  } catch (error) {
+    return NextResponse.json(
+      { error: 'Internal Server Error' },
+      { status: 500 }
+    );
+  }
+}
+
+export async function POST(request: NextRequest) {
+  try {
+    const res = await parseRequest(request);
+    const data = courseSchema.parse(res);
+    await prisma.course.create({ data: data });
+    return NextResponse.json({ data: data }, { status: 201 });
+  } catch (error) {
+    return NextResponse.json(
+      { error: 'Internal Server Error' },
+      { status: 500 }
+    );
+  }
+}
+
+export async function DELETE(request: NextRequest) {
+  try {
+  } catch (error) {}
+}
+
+export async function PUT(request: NextRequest) {
+  try {
+  } catch (error) {}
+}
+
+/* A helper function to determine whether the request is from the server or the client:
+return the request.json() if the request is from the client and else return the given arguments directly */
+
+async function parseRequest(request: NextRequest) {
```

Comment:
Should be unnecessary? Don't see where server would use POST

Reply:
I believe this was done to make the tests work?